docs/en/deploy/Running-Alluxio-Using-Vagrant.md

```diff
@@ -162,13 +247,216 @@ and find some files written into your S3 bucket by the above tests.
 
 From a node in the cluster, you can ssh to other nodes in the cluster without password with:
 
-{% include Running-Alluxio-on-EC2/ssh-other-node.md %}
+```bash
+$ ssh AlluxioWorker1
+```
 
-## Destroy the cluster
+### Destroy the cluster
 
 Under `deploy/vagrant` directory, you can run:
 
-{% include Running-Alluxio-on-EC2/destroy.md %}
+```bash
+$ ./destroy
+```
 
 to destroy the cluster that you created. Only one cluster can be created at a time. After the
 command succeeds, the EC2 instances are terminated.
+
+### Advanced Tips
+**Spot Instance**
+Using spot instances is a way to reduce EC2 cost. Spot instances are non-guaranteed instances which are priced with bidding.
+Note that spot instances may be taken away from you if someone bids more, and there are no more spot instances available.
+However, for short-term testing, spot instances are very appropriate, because it is rare that spot instances are taken from you.
+
+By default, the deploy scripts DO NOT use spot instances. Therefore, you have to enable deploy scripts to use spot instances.
+
+In order to enable spot instances, you have to modify the file: `deploy/vagrant/conf/ec2.yml`:
+
+    Spot_Price: “X.XX”
+
+For AWS EC2, the default underfs is S3. You need to sign into your [Amazon S3 console](http://aws.amazon.com/s3/), create a S3 bucket and write the bucket's name to the field `S3:Bucket` in `conf/ufs.yml`. To use other under storage systems, configure the field `Type` and the corresponding configurations in `conf/ufs.yml`.
+
+Now you can launch the Alluxio cluster with your chosen under filesystem in your chosen availability zone by running
+the script under `deploy/vagrant`:
+
+```bash
+$ ./create <number of machines> aws
+```
+
+Each node of the cluster runs an Alluxio worker, and the `AlluxioMaster` runs the Alluxio master.
+
+## Deploy with Vagrant on GCE
+
+### Prerequisites
+
+**Install Vagrant and the Google plugin**
```

Comment:
Can these prerequisites be a bulleted list?

Reply:
done